### PR TITLE
[stable10] Respect sharing.federation.allowHttpFallback in FederatedShareProvider

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
By default federated sharing should use https only - in selected cases a fallback to http is allowed.

## Related Issue
refs https://github.com/owncloud/core/issues/31194

## How Has This Been Tested?
1. have an owncloud instance available on http only
2. create federated share via https
3. falback to http is only allowed if the config switch is set to true

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1: code looks good

Mind adding test steps ? This likely requires some SSL web server ?

**Comment by patrickjahns** [comment]
I always see when trying to initiate a federated share to a http server - no matter if the option is set to true/false
image

File: apps/federatedfilesharing/lib/BackgroundJob/RetryJob.php [hunk 1]
```diff
@@ -74,7 +74,8 @@ public function __construct(Notifications $notifications = null) {
 				$addressHandler,
 				\OC::$server->getHTTPClientService(),
 				$discoveryManager,
-				\OC::$server->getJobList()
+				\OC::$server->getJobList(),
+				\OC::$server->getConfig()
```
Comment (PVince81) [inline apps/federatedfilesharing/lib/BackgroundJob/RetryJob.php]: so cron.php needs to be tested as well

**Review by PVince81 — APPROVED** [review]
:+1: code change with break makes sense

**Comment by patrickjahns** [comment]
Tested with http only receiving server:

With `'sharing.federation.allowHttpFallback' => false` I see this information
image

With `'sharing.federation.allowHttpFallback' => true` it allows for sharing

**Review by patrickjahns — APPROVED** [review]
LGTM - tested locally

**Pull request merged**